chain/jsonrpc/src/lib.rs

```diff
@@ -1330,23 +1354,26 @@ fn get_cors(cors_allowed_origins: &[String]) -> Cors {
 }
 
 #[get("/debug")]
-async fn debug_html() -> actix_web::Result<impl actix_web::Responder> {
-    Ok(HttpResponse::Ok().body(include_str!("../res/debug.html")))
+async fn debug_html(
+    handler: web::Data<JsonRpcHandler>,
+) -> actix_web::Result<impl actix_web::Responder> {
+    Ok(HttpResponse::Ok().body(debug_page_string!("debug.html", handler)))
 }
 
 #[get("/debug/pages/{page}")]
 async fn display_debug_html(
     path: web::Path<(String,)>,
+    handler: web::Data<JsonRpcHandler>,
 ) -> actix_web::Result<impl actix_web::Responder> {
     let page_name = path.into_inner().0;
 
```

Comment:
I'd defile `macro_rule!` right in this function, no need to make it crate-global

```
async fn display_debug_html(
    path: web::Path<(String,)>,
) -> actix_web::Result<impl actix_web::Responder> {
    let page_name = path.into_inner().0.as_str();
    macro_rules! get_html {
        ($(page:lit),*) => {
            match page_name { // local macros close over local vars
            $(
                $page => read($page).unwrap_or_else(include_str!(concat!("../res", $page)))
            )*
            _ => return Ok(HttpResponse::NotFound().finish())
            }
        };
    };
    let html = get_html!("last_blocks", "network_info", ...);
    Ok(HttpResponse::Ok().insert_header(header::ContentType::html()).body(html))
}
```

Reply:
There's one more in the other function, but I moved the macro to the file.